Assignment1/irisdata.py

```diff
@@ -0,0 +1,246 @@
+        data_df,
+        x="species",
+        y="sepal.length",
+        color="species",
+        title="Bar Plot of Sepal Length",
+    )
+    fig6 = px.bar(
+        data_df,
+        x="species",
+        y="sepal.width",
+        color="species",
+        title="Bar Plot of Sepal width",
+    )
+    fig7 = px.bar(
+        data_df,
+        x="species",
+        y="petal.length",
+        color="species",
+        title="Bar Plot of Petal Length",
+    )
+    fig8 = px.bar(
+        data_df,
+        x="species",
+        y="petal.width",
+        color="species",
+        title="Bar Plot of Petal Width",
+    )
+    fig5.show()
+    fig6.show()
+    fig7.show()
+    fig8.show()
+    # Histogram
+    fig9 = px.histogram(
+        data_df, x="sepal.length", color="species", title="Histogram of Sepal length"
+    )
+    fig10 = px.histogram(
+        data_df, x="sepal.width", color="species", title="Histogram of Sepal width"
+    )
+    fig11 = px.histogram(
+        data_df, x="petal.length", color="species", title="Histogram of Petal length"
+    )
+    fig12 = px.histogram(
+        data_df, x="petal.width", color="species", title="Histogram of Petal width"
+    )
+    fig9.show()
+    fig10.show()
+    fig11.show()
+    fig12.show()
+    # Transforming the data using standard scaler
+    from sklearn.preprocessing import StandardScaler
+
+    scaler = StandardScaler()
+    x = scaler.fit_transform(x)
+    # Encoding the output variable
+    species = data_df.iloc[:, 4].values  # converting output variable to numpy array
+    from sklearn.preprocessing import LabelEncoder
+
+    encoder = LabelEncoder()
+    species = encoder.fit_transform(species)  # Encoding the out put values
+    # Splitting the data to test and training sets
+    from sklearn.model_selection import train_test_split
+
+    x_train, x_test, y_train, y_test = train_test_split(
+        x, species, test_size=0.3, random_state=42
+    )
+    # Fitting Random forest against the data
+    from sklearn.ensemble import RandomForestClassifier  # importing the classifier
+
+    classifier = RandomForestClassifier()
+    classifier.fit(x_train, y_train)  # training the model
+    y_pred = classifier.predict(x_test)  # predicting the output using test data
+    print("Predicted output values")
+    print(y_pred)
+    from sklearn.metrics import accuracy_score, confusion_matrix
+
+    print("Random forest model Results:")
+    print("Confusion Matrix :")
+    print(confusion_matrix(y_pred, y_test))
+    print("Accuracy Score :")
+    print(accuracy_score(y_pred, y_test) * 100)  # calculating the accuracy of the model
+    # Fitting different models using a piple line
+    from sklearn.linear_model import LogisticRegression
+    from sklearn.pipeline import Pipeline
+    from sklearn.svm import SVC
+    from sklearn.tree import DecisionTreeClassifier
+
+    randomforestpipeline = Pipeline(
+        [("scaler", StandardScaler()), ("Classifier", RandomForestClassifier())]
+    )
```

Comment:
Nice! you tried different pipelines. Appreciate it!

Reply:
Thank You!